checks.d/yarn.py

```diff
@@ -0,0 +1,264 @@
+
+        self._yarn_cluster_metrics(resourcemanager_hostname, cluster_id, rm_address)
+        self._yarn_app_metrics(resourcemanager_hostname, rm_address)
+        self._yarn_node_metrics(resourcemanager_hostname, cluster_id, rm_address)
+
+    def _get_cluster_id(self, rm_address):
+
+        # Get the cluster info
+        info_json = self._rest_request_to_json(rm_address, INFO_URI)
+
+        if info_json:
+            if info_json['clusterInfo']:
+                if info_json['clusterInfo']['id']:
+                    return info_json['clusterInfo']['id']
+
+        return None
+
+
+    def _yarn_cluster_metrics(self, hostname, cluster_id, rm_address):
+        metrics_json = self._rest_request_to_json(rm_address, YARN_CLUSTER_METRICS_PATH)
+
+        if metrics_json:
+            keys = ['cluster_id:' + str(cluster_id)]
+
+            if metrics_json[YARN_CLUSTER_METRICS_ELEMENT] is not None:
+
+                yarn_metrics = metrics_json[YARN_CLUSTER_METRICS_ELEMENT]
+
+                self._set_yarn_metrics_from_json(keys, hostname, yarn_metrics, YARN_CLUSTER_METRICS)
+
+    def _yarn_app_metrics(self, hostname, rm_address):
+        metrics_json = self._rest_request_to_json(rm_address, YARN_APPS_PATH)
+
+        if metrics_json:
+
+            if metrics_json['apps'] is not None:
+                if metrics_json['apps']['app'] is not None:
+
+                    for app_json in metrics_json['apps']['app']:
+
+                        cluster_id = app_json['clusterId']
+                        app_id = app_json['id']
+
+                        keys = ['cluster_id:' + str(cluster_id)]
+                        keys.append('app_id:' + str(app_id))
+
+                        self._set_yarn_metrics_from_json(keys, hostname, app_json, YARN_APP_METRICS)
+
+
+    def _yarn_node_metrics(self, hostname, cluster_id, rm_address):
+        metrics_json = self._rest_request_to_json(rm_address, YARN_NODES_PATH)
+
+        if metrics_json:
+
+            if metrics_json['nodes'] is not None:
+                if metrics_json['nodes']['node'] is not None:
+
+                    keys = ['cluster_id:' + str(cluster_id)]
+
+                    for node_json in metrics_json['nodes']['node']:
+                        node_id = node_json['id']
+                        keys.append('node_id:' + str(node_id))
+
+                        self._set_yarn_metrics_from_json(keys, hostname, node_json, YARN_NODE_METRICS)
+
+    def _set_yarn_metrics_from_json(self, keys, hostname, metrics_json, yarn_metrics):
+        for status, metric in yarn_metrics.iteritems():
+            metric_name, metric_type = metric
+
+            if metrics_json[status] is not None:
+                self._set_metric(metric_name,
+                    metric_type,
+                    metrics_json[status],
+                    keys,
+                    hostname)
+
+
+    def _set_metric(self, metric_name, metric_type, value, tags=None, hostname=None, device_name=None):
```

Comment:
you shouldn't pass the hostname to your metrics as it will most likely be localhost.